applications/ConvectionDiffusionApplication/custom_utilities/embedded_mls_constraint_process.cpp

```diff
@@ -0,0 +1,437 @@
+    ReactivateElementsAndNodes();
+
+    // Apply extension constraints to negative nodes of split elements
+    ApplyExtensionConstraints(ext_op_map);
+}
+
+/* Protected functions ****************************************************/
+
+/* Private functions ****************************************************/
+   
+void EmbeddedMLSConstraintProcess::CalculateConformingExtensionBasis(NodesCloudMapType& rExtensionOperatorMap)
+{
+    // Get the MLS shape functions function
+    auto p_mls_sh_func = GetMLSShapeFunctionsFunction();
+
+    // Loop the elements to create the negative nodes MLS basis
+    for (auto& rElement : mpModelPart->Elements()) {
+        // Check if the element is split
+        const auto& r_geom = rElement.GetGeometry();
+        if (IsSplit(r_geom)) {
+            // Find the intersected element negative nodes
+            for (auto& r_node : r_geom) {
+                // Check whether node is negative (not ACTIVE) or already was added because of a previous element
+                const std::size_t found = rExtensionOperatorMap.count(&r_node);
+                if (r_node.IsNot(ACTIVE) && !found) {
+
+                    // Get the current negative node neighbours cloud
+                    Matrix cloud_nodes_coordinates;
+                    PointerVector<NodeType> cloud_nodes;
+                    SetNegativeNodeSupportCloud(r_node, cloud_nodes, cloud_nodes_coordinates);
+
+                    // Calculate the MLS basis in the current negative node
+                    Vector N_container;
+                    const array_1d<double,3> r_coords = r_node.Coordinates();
+                    const double mls_kernel_rad = CalculateKernelRadius(cloud_nodes_coordinates, r_coords);
+                    p_mls_sh_func(cloud_nodes_coordinates, r_coords, 1.01 * mls_kernel_rad, N_container);
+
+                    // Save the extension operator nodal data
+                    std::size_t n_cl_nod = cloud_nodes.size();
+                    CloudDataVectorType cloud_data_vector(n_cl_nod);
+                    for (std::size_t i_cl_nod = 0; i_cl_nod < n_cl_nod; ++i_cl_nod) {
+                        auto p_cl_node = cloud_nodes(i_cl_nod);
+                        auto i_data = std::make_pair(p_cl_node, N_container[i_cl_nod]);
+                        cloud_data_vector(i_cl_nod) = i_data;
+                    }
+
+                    auto ext_op_key_data = std::make_pair(&r_node, cloud_data_vector);
+                    rExtensionOperatorMap.insert(ext_op_key_data);
+                }
+            }
+        }
+    }
+}
+
+void EmbeddedMLSConstraintProcess::ApplyExtensionConstraints(NodesCloudMapType& rExtensionOperatorMap)
+{
+    // Initialize counter of master slave constraints
+    ModelPart::IndexType id = mpModelPart->NumberOfMasterSlaveConstraints()+1;
+    // Get variable to constrain 
+    const auto& r_var = KratosComponents<Variable<double>>::Get("TEMPERATURE");

```

Comment:
Here I was wondering, will it always be TEMPERATURE or should we make this more general?